tests/phpunit/tests/rest-api/rest-schema-validation.php

```diff
@@ -1253,4 +1253,252 @@ public function test_string_or_integer() {
 		$this->assertWPError( rest_validate_value_from_schema( 15.5, $schema ) );
 	}
 
+	/**
+	 * @ticket 51025
+	 *
+	 * @dataProvider data_any_of
+	 *
+	 * @param array $data
+	 * @param array $schema
+	 * @param bool $valid
+	 */
+	public function test_any_of( $data, $schema, $valid ) {
+		$is_valid = rest_validate_value_from_schema( $data, $schema );
+
+		if ( $valid ) {
+			$this->assertTrue( $is_valid );
+		} else {
+			$this->assertWPError( $is_valid );
+		}
+	}
+
+	/**
+	 * @return array
+	 */
+	public function data_any_of() {
+		$suites = json_decode( file_get_contents( __DIR__ . '/json_schema_test_suite/anyof.json' ), true );
+		$skip   = array(
+			'anyOf with boolean schemas, all true',
+			'anyOf with boolean schemas, some true',
+			'anyOf with boolean schemas, all false',
+			'anyOf with one empty schema',
+			'nested anyOf, to check validation semantics',
+		);
+
+		$tests = array();
+
+		foreach ( $suites as $suite ) {
+			if ( in_array( $suite['description'], $skip, true ) ) {
+				continue;
+			}
+
+			foreach ( $suite['tests'] as $test ) {
+				$tests[ $suite['description'] . ': ' . $test['description'] ] = array(
+					$test['data'],
+					$suite['schema'],
+					$test['valid'],
+				);
+			}
+		}
+
+		return $tests;
+	}
+
+	/**
+	 * @ticket 51025
+	 *
+	 * @dataProvider data_one_of
+	 *
+	 * @param array $data
+	 * @param array $schema
+	 * @param bool $valid
+	 */
+	public function test_one_of( $data, $schema, $valid ) {
+		$is_valid = rest_validate_value_from_schema( $data, $schema );
+
+		if ( $valid ) {
+			$this->assertTrue( $is_valid );
+		} else {
+			$this->assertWPError( $is_valid );
+		}
+	}
+
+	/**
+	 * @return array
+	 */
+	public function data_one_of() {
+		$suites = json_decode( file_get_contents( __DIR__ . '/json_schema_test_suite/oneof.json' ), true );
+		$skip   = array(
+			'oneOf with boolean schemas, all true',
+			'oneOf with boolean schemas, one true',
+			'oneOf with boolean schemas, more than one true',
+			'oneOf with boolean schemas, all false',
+			'oneOf with empty schema',
+			'nested oneOf, to check validation semantics',
+		);
+
+		$tests = array();
+
+		foreach ( $suites as $suite ) {
+			if ( in_array( $suite['description'], $skip, true ) ) {
+				continue;
+			}
+
+			foreach ( $suite['tests'] as $test ) {
+				$tests[ $suite['description'] . ': ' . $test['description'] ] = array(
+					$test['data'],
+					$suite['schema'],
+					$test['valid'],
+				);
+			}
+		}
+
+		return $tests;
+	}
+
+	/**
+	 * @ticket 51025
+	 *
+	 * @dataProvider data_combining_operation_error_message
+	 *
+	 * @param $data
+	 * @param $schema
+	 * @param $expected
+	 */
+	public function test_combining_operation_error_message( $data, $schema, $expected ) {
+		$is_valid = rest_validate_value_from_schema( $data, $schema );
```

Comment:
Can we specify a `param` here, I think it'll make the error messages a bit easier to read and understand.